subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_pdu.h

```diff
@@ -195,6 +195,11 @@ lll_adv_sync_data_latest_peek(const struct lll_adv_sync *const lll)
 	return lll_adv_pdu_latest_peek(&lll->data);
 }
 
+static inline struct pdu_adv *lll_adv_sync_data_curr_get(struct lll_adv_sync *lll)
+{
+	return (void *)lll->data.pdu[lll->data.first];
+}
```

Comment:
@cvinayak  will you be able to add more information here, as this function follows the same design as `struct pdu_adv *lll_adv_aux_data_curr_get(struct lll_adv_aux *lll)`

Reply:
The ADV PDU are double buffered. `first`  buffer remains persistent without being modified until the LLL execution consumes it, `first` is modified by LLL execution context and safely readable by both ULL and LLL. 

`last` is modified by ULL context and safely readable by both ULL and LLL.

LLL always peeks the latest in `last` index... i.e. it modifies `first` until it is `last`  index when dequeueing, then uses the data in the `last` index.

The `_curr_get()` is the naming used for the first data placed by ULL context that is not yet consumed by LLL. In the context of PAST only the AdvA is read back, which will anyway be same in case application is continuously updating data before LLL dequeues it, hence the name and its use is correct here.

Reply:
Is that documented anywhere? Just for future references

Reply:
I thought we had something, but seems it is not upstreamed, hence create this: https://github.com/zephyrproject-rtos/zephyr/issues/79643